flash/image/classification/data.py

```diff
@@ -11,206 +11,47 @@
-            self.set_state(ClassificationState(target_keys, classification_state.num_classes))
-
-        return result
-
-
-class ImageClassificationCSVInput(ImageClassificationDataFrameInput):
-    def load_data(
-        self,
-        csv_file: PATH_TYPE,
-        input_key: str,
-        target_keys: Optional[Union[str, List[str]]] = None,
-        root: Optional[PATH_TYPE] = None,
-        resolver: Optional[Callable[[Optional[PATH_TYPE], Any], PATH_TYPE]] = None,
-    ) -> List[Dict[str, Any]]:
-        data_frame = read_csv(csv_file)
-        if root is None:
-            root = os.path.dirname(csv_file)
-        return super().load_data(data_frame, input_key, target_keys, root, resolver)
-
-
-class ImageClassificationInputTransform(InputTransform):
-    """Preprocssing of data of image classification."""
-
-    def __init__(
-        self,
-        train_transform: Optional[Dict[str, Callable]] = None,
-        val_transform: Optional[Dict[str, Callable]] = None,
-        test_transform: Optional[Dict[str, Callable]] = None,
-        predict_transform: Optional[Dict[str, Callable]] = None,
-        image_size: Tuple[int, int] = (196, 196),
-        deserializer: Optional[Deserializer] = None,
-    ):
-        self.image_size = image_size
-
-        super().__init__(
-            train_transform=train_transform,
-            val_transform=val_transform,
-            test_transform=test_transform,
-            predict_transform=predict_transform,
-            inputs={
-                InputFormat.FIFTYONE: ImageClassificationFiftyOneInput,
-                InputFormat.FILES: ImageClassificationFilesInput,
-                InputFormat.FOLDERS: ImageClassificationFolderInput,
-                InputFormat.NUMPY: ImageClassificationNumpyInput,
-                InputFormat.TENSORS: ImageClassificationTensorInput,
-                InputFormat.DATAFRAME: ImageClassificationDataFrameInput,
-                InputFormat.CSV: ImageClassificationCSVInput,
-                InputFormat.LABELSTUDIO: LabelStudioImageClassificationInput,
-            },
-            deserializer=deserializer or ImageDeserializer(),
-            default_input=InputFormat.FILES,
-        )
-
-    def get_state_dict(self) -> Dict[str, Any]:
-        return {**self.transforms, "image_size": self.image_size}
-
-    @classmethod
-    def load_state_dict(cls, state_dict: Dict[str, Any], strict: bool = False):
-        return cls(**state_dict)
-
-    def default_transforms(self) -> Optional[Dict[str, Callable]]:
-        return default_transforms(self.image_size)
-
-    def train_default_transforms(self) -> Optional[Dict[str, Callable]]:
-        return train_default_transforms(self.image_size)
-
-
 class ImageClassificationData(DataModule):
     """Data module for image classification tasks."""
 
+    input_transforms_registry = FlashRegistry("input_transforms")
     input_transform_cls = ImageClassificationInputTransform
```

Comment:
Is this needed? `ImageClassificationInputTransform` is used explicitly by the classmethods

Reply:
Yes, it is injected in the dataset and used to create the transform. This would be used if transform are registered and a matching key is passed.